src/pkg/extensions/mock_extensions.go

```diff
@@ -14,60 +14,61 @@ import (
 var _ io.ReadCloser = &MockExtension{}
 
 type MockExtension struct {
-	numBytes    int
-	crc32       uint32
-	info        details.ItemInfo
-	extInfo     *details.ExtensionInfo
-	innerRc     io.ReadCloser
-	ctx         context.Context
-	failOnRead  bool
-	failOnClose bool
+	NumBytes    int64
+	Crc32       uint32
+	Info        details.ItemInfo
+	ExtInfo     *details.ExtensionInfo
+	InnerRc     io.ReadCloser
+	Ctx         context.Context
```

Comment:
Mock is just a replica of how actual extensions will work. We are passing `ctx` through `CreateItemExtension` which is eventually stored here.

Not sure if there is an easy way to provide it as a runtime func param, since this is an `io.ReadCloser` wrapper